src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_CAPACITY = 10;
+    private static final int HALF_DIVIDER_INDEX = 2;
+    private int size;
+    private Object[] elements;
```

Comment:
we can do better than Object here, take a look at what do we have just a few lines above 😉